tasks/deploy.rb

```diff
@@ -0,0 +1,37 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'json'
+
+params       = JSON.parse(STDIN.read)
+
+# Munge items into arrays of things to deploy
+environments = params['environments'] || params['environment']
+environments = environments.split(',') if environments
+
+modules      = params['modules'] || params['module']
+modules      = modules.split(',') if modules
+
+messages   = []
+exitstatus = 0
+if environments
+  environments.each do |env|
+    messages << "Deploying environment #{env}."
+    `/opt/puppetlabs/bin/r10k deploy environment #{env} --puppetfile`
+    exitstatus += $?.exitstatus
+  end
+end
+
+if modules
+  list = modules.join(',')
+  messages << "Deploying modules #{list}."
+  `/opt/puppetlabs/bin/r10k deploy module #{list}`
```

Comment:
can we rely on a proper PATH variable here? Otherwise we need a solution to set the path or something similar. This will fail on systems where r10k isn't installed into puppet.

Reply:
I don't have a solid answer for that, but relying on `$PATH` seemed to work for me, so I took out the fully qualified path.